frontend/src/MuiForms/Metrics.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+  Button,
+  Paper,
+  Stack,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Typography,
+} from '@mui/material'
+import { useTheme } from '@mui/material/styles'
+import { FormContextType } from '@rjsf/utils'
+import _ from 'lodash-es'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import MetricItem from 'src/MuiForms/MetricItem'
+import { isValidNumber } from 'utils/stringUtils'
+import { v4 as uuidv4 } from 'uuid'
+
+export interface MetricValue {
+  name: string
+  value: number | string
+}
+
+interface Id {
+  id: string
+}
+
+export type MetricValueWithId = MetricValue & Id
+
+interface MetricsProps {
+  onChange: (newValue: MetricValue[]) => void
+  value: MetricValue[]
+  label: string
+  formContext?: FormContextType
+  required?: boolean
+}
+
+export default function Metrics({ onChange, value, label, formContext, required }: MetricsProps) {
+  const [metricsWithIds, setMetricsWithIds] = useState<MetricValueWithId[]>([])
+  const theme = useTheme()
+
+  useEffect(() => {
+    const updatedMetricsWithIds = value.map((metric) => ({
+      ...metric,
+      id: uuidv4(),
+    }))
+    setMetricsWithIds(updatedMetricsWithIds)
+  }, [value])
+
+  const handleChange = useCallback(
+    (newValues: MetricValue[]) => {
+      onChange(newValues)
+    },
+    [onChange],
+  )
+
+  const handleMetricItemOnChange = useCallback(
+    (updatedMetricItem: MetricValueWithId) => {
+      const updatedMetricArray = _.cloneDeep(metricsWithIds)
+      const index = metricsWithIds.findIndex((metric) => metric.id === updatedMetricItem.id)
+      updatedMetricArray[index] = updatedMetricItem
+      onChange(
+        updatedMetricArray.map((metric) => ({
+          name: metric.name,
+          value: isValidNumber(metric.value as string) ? parseInt(metric.value as string) : metric.value,
```

Comment:
`isValidNumber('1e+12')` is truthy as `1e+12` is casted to a `Number` however `parseInt('1e+12')` is `1`. So if a user enters a large number it'll change back to a small number.